### PR TITLE
misspell: Add locale setting to misspell

### PR DESCRIPTION
**Pull request by purpleidea**
This adds a --misspell-locale option to gometalinter. It passes the
desired locale setting to misspell so that you can check for common
typos like "color" which are obviously supposed to read "colour". ;)
Unfortunately there is no Canadian linter at this time, so UK will have
to do for now.

Example:

gometalinter --disable-all --enable=misspell --misspell-locale=UK

### CODE REVIEW

**Comment by purpleidea** [comment]
Please note: I did not test this patch yet, so please give it a once over before you merge.

Thanks!

**Comment by alecthomas** [comment]
The code does not compile. See the CI.

**Comment by purpleidea** [comment]
@alecthomas My bad, I had a one char typo, since I didn't want to setup a custom GOPATH to test this :P (just use git master instead of vendoring please!!)

In any case, patch pushed (`StringVar` not `StringsVar`)

No idea why travis is failing now (logs not visible)-- this builds on my machine.

**Comment by alecthomas** [comment]
The regression tests are failing. I suspect because the default empty locale results in `misspell -j 1 --locale ` (note the lack of argument to `--locale`).

```
--- FAIL: TestMisSpell (1.21s)
	Error Trace:	support.go:51
			misspell_test.go:15
	Error:      	Not equal: 
	            	expected: regressiontests.Issues{regressiontests.Issue{Linter:"misspell", Severity:"warning", Path:"test.go", Line:2, Col:7, Message:"\"langauge\" is a misspelling of \"language\""}, regressiontests.Issue{Linter:"misspell", Severity:"warning", Path:"test.go", Line:3, Col:9, Message:"\"langauge\" is a misspelling of \"language\""}}
	            	actual: regressiontests.Issues{}
	            	
	            	Diff:
	            	--- Expected
	            	+++ Actual
	            	@@ -1,4 +1,2 @@
	            	-(regressiontests.Issues) (len=2) {
	            	- (regressiontests.Issue) test.go:2:7:warning: "langauge" is a misspelling of "language" (misspell),
	            	- (regressiontests.Issue) test.go:3:9:warning: "langauge" is a misspelling of "language" (misspell)
	            	+(regressiontests.Issues) {
	            	 }
```

**Comment by purpleidea** [comment]
@alecthomas I've digged a little bit, but your test infra is sufficiently complex that it's hard to follow.
I don't understand how this is the case since passing an empty locale to misspell should simply *not* add US or UK dictionaries, and instead use only the main dictionary. I've checked that the `language` misspelling is included in that dictionary.

I've added quotes around the option in case it needs this to interpret an empty string. If this doesn't work, I'm afraid I'll have to abandon the patch if you can't suggest a proper solution.

Thanks!

**Comment by alecthomas** [comment]
> I've added quotes around the option in case it needs this to interpret an empty string.

That is exactly the case, and what I was referring to.

**Comment by purpleidea** [comment]
\o/

**Comment by purpleidea** [comment]
Hey so I wrote patch 9af9a4364096fb170d968120f019fd9ad7234cd2 yet you merged 94ffa0f81f02e7bd6a9b33c2549d4a4de73d0ee6 ... what gives?

**Comment by alecthomas** [comment]
It's a squash merge.

**Comment by purpleidea** [comment]
Just fast-forward when you can :) It's more pro!

**Comment by alecthomas** [comment]
I'll take it under advisement.
